docs/image-build-and-customization.md

```diff
@@ -0,0 +1,283 @@
+
+1. **Specifications for the image** - these are defined as options to the command itself. Available specification options can be viewed by calling the command's help message: `elemental3 build -h`.
+2. **Information about the desired state of the image** - this is done through the definition of a configuration directory that the customer creates beforehand. For more information on the directory itself, refer to the [Configuration Directory Guide](configuration-directory.md).
+
+We recommend you familiarize yourself with both points before attempting to build your first image.
+
+### Limitations
+
+Currently the image build and customization process has the following limitations:
+
+1. Supports building images only for `x86_64` platforms.
+2. Supports building only `RAW` type images.
+3. Supports building images for non air-gapped environments only.
+3. Does not support building Linux-only images. Customers looking to build Linux-only images should refer to the [Linux-only Image](linux-only-image.md) guide.
+
+Elemental is in active development and much of these limitations are on our roadmap and **will** be fixed.
+
+### Usage
+
+Below you can find the minimum set of options that the command requires:
+
+```shell
+sudo elemental3 build --image-type <value> --config-dir <path>
+```
+Unless configured otherwise, after execution, the process will create a directory called `_build` which will contain the ready to boot image.
+
+> **NOTE:** If you have specified either the `--build-dir` or `--output` options, your build directory and/or image name will be different.
+
+For more information on what the `_build` directory is about, refer to the [Build Directory Overview](#build-directory-overview) section.
+
+For an overview of the workflow that this command goes through, refer to the [Build and Customization Process Overview](#build-and-customization-process-overview) section.
+
+For a real-life example usage for the command, refer to the [Example](#example) section.
+
+### Build Directory Overview
+
+A build directory is created for each execution of the `elemental3 build` command. Apart from the actual built image, this build directory will also hold all the files and sub-directories that where used during the build process.
+
+Expanded, the aforementioned build directory holds the following files/sub-directories:
+
+```shell
+_build/
+├── build-<timestamp>
+│   ├── config.sh
+│   ├── overlays
+│   └── release-manifests
+└── image-<timestamp>.raw
+```
+*Files:*
+* config.sh - script responsible for applying configurations to the operating system during installation.
+* image-<timestamp>.raw - the built image. This file will be present in the build directory only if the `--output` option was not specified.
+
+*Directories:*
+* overlays - resources under this directory will be directly overlayed onto the booted operating system.
+* release-manifests - holds any release manifests that were pulled from an OCI image.
```

Comment:
Sorry if I'm missing something -- do we have this if we're using a plain file reference?